### PR TITLE
Reworked EXO Unit Tests to use 1 JSON

### PR DESCRIPTION
**Pull request by dagarwal-mitre**
#  #
  

## 🗣 Description ##

Create central JSON config that can be pulled and modified slightly for each unit test
Closing #1111 
## 💭 Motivation and context ##

Reduce copy and paste errors and file size. Reduce the difficulty of writing unit tests and PRs

## 🧪 Testing ##

- [x] All EXO unit tests pass
- [x] Unit tests utilize test assertions for debugging
- [x] RunUnitTests.ps1 passes all the new utility files

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] PR targets the correct parent branch (e.g., main or release-name) for merge.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] Changes are sized such that they do not touch excessive number of files.
- [x] *All* future TODOs are captured in issues, which are referenced in code comments.
- [x] These code changes follow the ScubaGear [content style guide](https://github.com/cisagov/ScubaGear/blob/main/CONTENTSTYLEGUIDE.md).
- [x] Related issues these changes resolve are linked preferably via [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [x] All relevant type-of-change labels added.
- [x] All relevant project fields are set.
- [x] All relevant repo and/or project documentation updated to reflect these changes.
- [x] Unit tests added/updated to cover PowerShell and Rego changes.
- [x] Functional tests added/updated to cover PowerShell and Rego changes.
- [x] All relevant functional tests passed.
- [ ] All automated checks (e.g., linting, static analysis, unit/smoke tests) passed.

## ✅ Pre-merge checklist ##

- [ ] PR passed smoke test check.
- [ ] Feature branch has been rebased against changes from parent branch, as needed

  Use `Rebase branch` button below or use [this](https://www.digitalocean.com/community/tutorials/how-to-rebase-and-update-a-pull-request) reference to rebase from the command line.
- [ ] Resolved all merge conflicts on branch
- [ ] Notified merge coordinator that PR is ready for merge via comment mention

## ✅ Post-merge checklist ##

- [ ] Feature branch deleted after merge to clean up repository.
- [ ] Verified that all checks pass on parent branch (e.g., main or release-name) after merge.

### CODE REVIEW

File: PowerShell/ScubaGear/Testing/Unit/Rego/EXO/EXOConfig_06_test.rego [hunk 1]
```diff
@@ -9,33 +9,15 @@ import data.utils.key.PASS
...
-            {
-                "Domains": [
-                    "*:ContactsSharing",
-                    "domain1:CalendarSharingFreeBusyDetail"
-                ],
-                "Name": "A"
-            }
-        ]
-    }
+    SharingPolicy1 := json.patch(SharingPolicy, [{"op": "add", "path": "Domains", "value":["*:ContactsSharing", "domain1:CalendarSharingFreeBusyDetail"]}])
```
Comment (adhilto) [inline PowerShell/ScubaGear/Testing/Unit/Rego/EXO/EXOConfig_06_test.rego:18]: I know this is just a warning in Regal not an error, still, I'd recommend breaking up this line to get rid of the "line too long" warning. Something like this should work:
```
SharingPolicy1 := json.patch(SharingPolicy, [{
    "op": "add",
    "path": "Domains",
    "value": ["*:ContactsSharing", "domain1:CalendarSharingFreeBusyDetail"]
}])
```

File: PowerShell/ScubaGear/Testing/Unit/Rego/EXO/EXOConfig_06_test.rego [hunk 2]
```diff
@@ -52,33 +34,15 @@ test_Domains_Contacts_Incorrect if {
...
-            {
-                "Domains": [
-                    "*:CalendarSharingFreeBusyDetail",
-                    "domain1:ContactsSharing"
-                ],
-                "Name": "A"
-            }
-        ]
-    }
+    SharingPolicy1 := json.patch(SharingPolicy, [{"op": "add", "path": "Domains", "value":["*:CalendarSharingFreeBusyDetail", "domain1:ContactsSharing"]}])
```
Comment (adhilto) [inline PowerShell/ScubaGear/Testing/Unit/Rego/EXO/EXOConfig_06_test.rego:43]: Again, I'd recommend breaking down this line:
```
SharingPolicy1 := json.patch(SharingPolicy, [{
    "op": "add",
    "path": "Domains",
    "value": ["*:CalendarSharingFreeBusyDetail", "domain1:ContactsSharing"]
}])
```

File: PowerShell/ScubaGear/Testing/Unit/Rego/EXO/EXOConfig_04_test.rego [hunk 3]
```diff
@@ -116,72 +64,34 @@ test_Rdata_Incorrect_V3 if {
...
-                "rdata": [
-                    "spf1 "
-                ],
-                "domain": "test.name"
-            }
-        ]
-    }
+    SPFRecord := json.patch(SpfRecords, [{"op": "add", "path": "rdata", "value": ["spf1 "]},
+                                        {"op": "add", "path": "domain", "value": "test.name"}])
+    DmarcRecord1 := json.patch(DmarcRecords, [{"op": "add", "path": "rdata", "value": ["v=DMARC1; p=none; mailto:reports@dmarc.cyber.dhs.gov mailto:jsmith@dhs.gov mailto:jsomething@dhs.gov"]}])
```
Comment (adhilto) [inline PowerShell/ScubaGear/Testing/Unit/Rego/EXO/EXOConfig_04_test.rego:79]: Recommend breaking down this long line

File: PowerShell/ScubaGear/Testing/Unit/Rego/EXO/EXOConfig_04_test.rego [hunk 4]
```diff
@@ -116,72 +64,34 @@ test_Rdata_Incorrect_V3 if {
...
-                "rdata": [
-                    "spf1 "
-                ],
-                "domain": "test.name"
-            }
-        ]
-    }
+    SPFRecord := json.patch(SpfRecords, [{"op": "add", "path": "rdata", "value": ["spf1 "]},
+                                        {"op": "add", "path": "domain", "value": "test.name"}])
+    DmarcRecord1 := json.patch(DmarcRecords, [{"op": "add", "path": "rdata", "value": ["v=DMARC1; mailto:reports@dmarc.cyber.dhs.gov mailto:jsmith@dhs.gov mailto:jsomething@dhs.gov"]}])
```
Comment (adhilto) [inline PowerShell/ScubaGear/Testing/Unit/Rego/EXO/EXOConfig_04_test.rego:91]: Long line

File: PowerShell/ScubaGear/Testing/Unit/Rego/EXO/EXOConfig_04_test.rego [hunk 5]
```diff
@@ -192,94 +102,48 @@ test_Rdata_Incorrect_V5 if {
...
-                "rdata": [
-                    "spf1 "
-                ],
-                "domain": "test.name"
-            }
-        ]
-    }
+    SPFRecord := json.patch(SpfRecords, [{"op": "add", "path": "rdata", "value": ["spf1 "]},
+                                        {"op": "add", "path": "domain", "value": "test.name"}])
+    DmarcRecord1 := json.patch(DmarcRecords, [{"op": "add", "path": "rdata", "value": ["v=DMARC1; p=reject; pct=100; rua=mailto:reports@wrong.address"]}])
```
Comment (adhilto) [inline PowerShell/ScubaGear/Testing/Unit/Rego/EXO/EXOConfig_04_test.rego:130]: Long line

File: PowerShell/ScubaGear/Testing/Unit/Rego/EXO/EXOConfig_04_test.rego [hunk 6]
```diff
@@ -292,236 +156,113 @@ test_DMARCReport_Incorrect_V3 if {
...
-                "rdata": [
-                    "spf1 "
-                ],
-                "domain": "test.name"
-            }
-        ]
-    }
+    SPFRecord := json.patch(SpfRecords, [{"op": "add", "path": "rdata", "value": ["spf1 "]},
+                                        {"op": "add", "path": "domain", "value": "test.name"}])
+    DmarcRecord1 := json.patch(DmarcRecords, [{"op": "add", "path": "rdata", "value": ["v=DMARC1; p=reject; pct=100; rua=mailto:reports@dmarc.cyber.dhs.gov"]}])
```
Comment (adhilto) [inline PowerShell/ScubaGear/Testing/Unit/Rego/EXO/EXOConfig_04_test.rego:185]: Long line

File: PowerShell/ScubaGear/Testing/Unit/Rego/EXO/EXOConfig_04_test.rego [hunk 7]
```diff
@@ -292,236 +156,113 @@ test_DMARCReport_Incorrect_V3 if {
...
-                "rdata": [
-                    "spf1 "
-                ],
-                "domain": "test.name"
-            }
-        ]
-    }
+    SPFRecord := json.patch(SpfRecords, [{"op": "add", "path": "rdata", "value": ["spf1 "]},
+                                        {"op": "add", "path": "domain", "value": "test.name"}])
+    DmarcRecord1 := json.patch(DmarcRecords, [{"op": "add", "path": "rdata", "value": ["v=DMARC1; p=reject; pct=100; rua=mailto:reports@dmarc.cyber.dhs.gov, test@exo.com"]}])
```
Comment (adhilto) [inline PowerShell/ScubaGear/Testing/Unit/Rego/EXO/EXOConfig_04_test.rego:198]: Long line

File: PowerShell/ScubaGear/Testing/Unit/Rego/EXO/EXOConfig_04_test.rego [hunk 8]
```diff
@@ -292,236 +156,113 @@ test_DMARCReport_Incorrect_V3 if {
...
-                ],
-                "domain": "example.com"
-            }
-        ]
-    }
+    SPFRecord := json.patch(SpfRecords, [{"op": "add", "path": "rdata", "value": ["spf1 "]},
+                                        {"op": "add", "path": "domain", "value": "test.name"}])
+    SPFRecord1 := json.patch(SpfRecords, [{"op": "add", "path": "rdata", "value": ["spf1 "]},
+                                        {"op": "add", "path": "domain", "value": "example.com"}])
+    DmarcRecord1 := json.patch(DmarcRecords, [{"op": "add", "path": "rdata", "value": ["v=DMARC1; p=reject; pct=100; rua=mailto:reports@dmarc.cyber.dhs.gov, test@test.name ruf=test2@test.name"]}])
```
Comment (adhilto) [inline PowerShell/ScubaGear/Testing/Unit/Rego/EXO/EXOConfig_04_test.rego:226]: Long line

File: PowerShell/ScubaGear/Testing/Unit/Rego/EXO/EXOConfig_04_test.rego [hunk 9]
```diff
@@ -292,236 +156,113 @@ test_DMARCReport_Incorrect_V3 if {
...
+    DmarcRecord2 := json.patch(DmarcRecords, [{"op": "add", "path": "rdata", "value": ["v=DMARC1; p=reject; pct=100; rua=mailto:reports@dmarc.cyber.dhs.gov"]},
```
Comment (adhilto) [inline PowerShell/ScubaGear/Testing/Unit/Rego/EXO/EXOConfig_04_test.rego:227]: Long line

File: PowerShell/ScubaGear/Testing/Unit/Rego/EXO/EXOConfig_04_test.rego [hunk 10]
```diff
@@ -292,236 +156,113 @@ test_DMARCReport_Incorrect_V3 if {
...
-                "domain": "example.com"
-            }
-        ]
-    }
+    SPFRecord := json.patch(SpfRecords, [{"op": "add", "path": "rdata", "value": ["spf1 "]},
+                                        {"op": "add", "path": "domain", "value": "test.name"}])
+    SPFRecord1 := json.patch(SpfRecords, [{"op": "add", "path": "rdata", "value": ["spf1 "]},
+                                        {"op": "add", "path": "domain", "value": "example.com"}])
+    DmarcRecord1 := json.patch(DmarcRecords, [{"op": "add", "path": "rdata", "value": ["v=DMARC1; p=reject; pct=100; rua=mailto:reports@dmarc.cyber.dhs.gov, test@test.name ruf=test2@test.name"]}])
+    DmarcRecord2 := json.patch(DmarcRecords, [{"op": "add", "path": "rdata", "value": ["v=DMARC1; p=reject; pct=100; rua=mailto:reports@dmarc.cyber.dhs.gov; ruf=test@exo.com"]},
```
Comment (adhilto) [inline PowerShell/ScubaGear/Testing/Unit/Rego/EXO/EXOConfig_04_test.rego:244]: Long line

**Review by adhilto — CHANGES_REQUESTED** [review]
Overall looks great. The one thing I'd like you to change: there were some really long lines that causing Regal warnings. In an effort to keep things clean and follow best practice, I'd recommend breaking those down. I provided a few examples of how you could do this, it should be very straightforward.

**Comment by dagarwal-mitre** [comment]
@nanda-katikaneni this pr is ready to merge

**Pull request merged**